### PR TITLE
Add test coverage for `lib` files

### LINKED ISSUE #463: Improve unit tests for files in the `lib` directory
## Is your feature request related to a problem? Please describe.
Most of the files in the `lib` directory are missing significant chunks of unit test coverage.

## Describe the solution you'd like
Add or enhance unit tests to improve coverage.

## Additional context
Coverage for this directory
```
Name                                                                         Stmts   Miss  Cover   Missing
----------------------------------------------------------------------------------------------------------
bag_transfer/lib/RAC_CMD.py                                                     68     32    53%   7-18, 25-26, 50, 54, 65-67, 76-78, 83-95, 99-102
bag_transfer/lib/__init__.py                                                     0      0   100%
bag_transfer/lib/bag_checker.py                                                174     42    76%   30-31, 33-34, 37-39, 47-48, 63-64, 71-75, 89-93, 96-100, 103-107, 110-114, 117-121, 124-128, 132, 169, 179-181, 189-191, 216-218, 237
bag_transfer/lib/cleanup.py                                                     18      3    83%   22-24
bag_transfer/lib/clients.py                                                     26     18    31%   11-18, 23-34, 37-49
bag_transfer/lib/cron.py                                                        91     35    62%   59-64, 69-87, 104-106, 123-177
bag_transfer/lib/files_helper.py                                               196     59    70%   64-66, 72, 76, 94-96, 100-111, 115-127, 144, 154-155, 166-167, 180-181, 203-204, 211-215, 219-221, 226-230, 237, 255-256, 268-269, 274-277
bag_transfer/lib/log_print.py                                                   25      7    72%   12-15, 19, 25, 37
bag_transfer/lib/mailer.py                                                      48     14    71%   22-23, 55-57, 63-72, 107-125
bag_transfer/lib/transfer_routine.py                                           269     76    72%   55-57, 62-63, 70, 82-83, 90-91, 94, 97, 100, 103, 154-155, 168, 177, 180-181, 187, 192-200, 286, 292, 299-300, 313-347, 358-371, 374-375, 399, 407, 410-413, 417-419, 423, 427-429
bag_transfer/lib/view_helpers.py                                                12      6    50%   2-5, 11, 13
bag_transfer/lib/virus_scanner.py                                               27     10    63%   21-32
```

### LINKED ISSUE #465: Replace custom ArchivesSpaceClient with ArchivesSnake
## Is your feature request related to a problem? Please describe.
There is a custom ArchivesSpaceClient in Aurora. This code could probably be replaced by using ArchivesSnake.

## Describe the solution you'd like
What the title says. May need to tweak unit tests.

### PR DESCRIPTION
**Pull request by helrond**
Add test coverage for files in the `lib` directory. Significantly, this includes the `BagChecker` and `TransferRoutine` classes. Also implements ArchivesSnake instead of our custom client.

Fixes #463 
Fixes #465

### CODE REVIEW

**Review by p-galligan — COMMENTED** [review]
I'm getting a _ton_ of errors from the tests on this. The test suite eventually passes, but it's finding something weird.

WARNING:bagit:bag-info.txt exists in manifest but was not found on filesystem
WARNING:bagit:bag-info.txt exists in manifest but was not found on filesystem
WARNING:bagit:bag-info.txt exists in manifest but was not found on filesystem
WARNING:bagit:bag-info.txt exists in manifest but was not found on filesystem
ERROR:root:/data/tmp/missing_description: Required tag 'Internal-Sender-Description' is not present in bag-info.txt. Set 'ignore_baginfo_tag_case' to True if you wish to ignore tag case.
ERROR:root:/data/tmp/missing_description: Required tag 'Internal-Sender-Description' is not present in bag-info.txt. Set 'ignore_baginfo_tag_case' to True if you wish to ignore tag case.
ERROR:root:/data/tmp/missing_description: Required tag 'Internal-Sender-Description' is not present in bag-info.txt. Set 'ignore_baginfo_tag_case' to True if you wish to ignore tag case.
ERROR:root:/data/tmp/missing_description: Required tag 'Internal-Sender-Description' is not present in bag-info.txt. Set 'ignore_baginfo_tag_case' to True if you wish to ignore tag case.
ERROR:root:/data/tmp/missing_record_type: Required tag 'Record-Type' is not present in bag-info.txt. Set 'ignore_baginfo_tag_case' to True if you wish to ignore tag case.
ERROR:root:/data/tmp/missing_record_type: Required tag 'Record-Type' is not present in bag-info.txt. Set 'ignore_baginfo_tag_case' to True if you wish to ignore tag case.
ERROR:root:/data/tmp/missing_record_type: Required tag 'Record-Type' is not present in bag-info.txt. Set 'ignore_baginfo_tag_case' to True if you wish to ignore tag case.
ERROR:root:/data/tmp/missing_record_type: Required tag 'Record-Type' is not present in bag-info.txt. Set 'ignore_baginfo_tag_case' to True if you wish to ignore tag case.
ERROR:root:/data/tmp/missing_source_organization: Required tag 'Source-Organization' is not present in bag-info.txt. Set 'ignore_baginfo_tag_case' to True if you wish to ignore tag case.
ERROR:root:/data/tmp/missing_source_organization: Required tag 'Source-Organization' is not present in bag-info.txt. Set 'ignore_baginfo_tag_case' to True if you wish to ignore tag case.
ERROR:root:/data/tmp/missing_source_organization: Required tag 'Source-Organization' is not present in bag-info.txt. Set 'ignore_baginfo_tag_case' to True if you wish to ignore tag case.
ERROR:root:/data/tmp/missing_source_organization: Required tag 'Source-Organization' is not present in bag-info.txt. Set 'ignore_baginfo_tag_case' to True if you wish to ignore tag case.
ERROR:root:/data/tmp/missing_title: Required tag 'Title' is not present in bag-info.txt. Set 'ignore_baginfo_tag_case' to True if you wish to ignore tag case.
ERROR:root:/data/tmp/missing_title: Required tag 'Title' is not present in bag-info.txt. Set 'ignore_baginfo_tag_case' to True if you wish to ignore tag case.
ERROR:root:/data/tmp/missing_title: Required tag 'Title' is not present in bag-info.txt. Set 'ignore_baginfo_tag_case' to True if you wish to ignore tag case.
ERROR:root:/data/tmp/missing_title: Required tag 'Title' is not present in bag-info.txt. Set 'ignore_baginfo_tag_case' to True if you wish to ignore tag case.
ERROR:root:/data/tmp/repeating_record_type: Required tag 'Record-Type' is present in bag-info.txt but does not have an allowed value ('['grant records', 'annual reports']').
ERROR:root:/data/tmp/repeating_record_type: Required tag 'Record-Type' is present in bag-info.txt but does not have an allowed value ('['grant records', 'annual reports']').
ERROR:root:/data/tmp/repeating_record_type: Required tag 'Record-Type' is present in bag-info.txt but does not have an allowed value ('['grant records', 'annual reports']').
ERROR:root:/data/tmp/repeating_record_type: Required tag 'Record-Type' is present in bag-info.txt but does not have an allowed value ('['grant records', 'annual reports']').
ERROR:root:/data/tmp/repeating_source_organization: Required tag 'Source-Organization' is present in bag-info.txt but does not have an allowed value ('['Ford Foundation', 'Rockefeller Foundation']').
ERROR:root:/data/tmp/repeating_source_organization: Required tag 'Source-Organization' is present in bag-info.txt but does not have an allowed value ('['Ford Foundation', 'Rockefeller Foundation']').
ERROR:root:/data/tmp/repeating_source_organization: Required tag 'Source-Organization' is present in bag-info.txt but does not have an allowed value ('['Ford Foundation', 'Rockefeller Foundation']').
ERROR:root:/data/tmp/repeating_source_organization: Required tag 'Source-Organization' is present in bag-info.txt but does not have an allowed value ('['Ford Foundation', 'Rockefeller Foundation']').
ERROR:root:/data/tmp/repeating_title: Nonrepeatable tag 'Title' occurs 2 times in bag-info.txt.
ERROR:root:/data/tmp/repeating_title: Nonrepeatable tag 'Title' occurs 2 times in bag-info.txt.
ERROR:root:/data/tmp/repeating_title: Nonrepeatable tag 'Title' occurs 2 times in bag-info.txt.
ERROR:root:/data/tmp/repeating_title: Nonrepeatable tag 'Title' occurs 2 times in bag-info.txt.
ERROR:root:/data/tmp/unauthorized_record_type: Required tag 'Record-Type' is present in bag-info.txt but does not have an allowed value ('grants').
ERROR:root:/data/tmp/unauthorized_record_type: Required tag 'Record-Type' is present in bag-info.txt but does not have an allowed value ('grants').
ERROR:root:/data/tmp/unauthorized_record_type: Required tag 'Record-Type' is present in bag-info.txt but does not have an allowed value ('grants').
ERROR:root:/data/tmp/unauthorized_record_type: Required tag 'Record-Type' is present in bag-info.txt but does not have an allowed value ('grants').
ERROR:root:/data/tmp/unauthorized_source_organization: Required tag 'Source-Organization' is present in bag-info.txt but does not have an allowed value ('Kinks Community Foundation').
ERROR:root:/data/tmp/unauthorized_source_organization: Required tag 'Source-Organization' is present in bag-info.txt but does not have an allowed value ('Kinks Community Foundation').
ERROR:root:/data/tmp/unauthorized_source_organization: Required tag 'Source-Organization' is present in bag-info.txt but does not have an allowed value ('Kinks Community Foundation').
ERROR:root:/data/tmp/unauthorized_source_organization: Required tag 'Source-Organization' is present in bag-info.txt but does not have an allowed value ('Kinks Community Foundation').

**Comment by helrond** [comment]
@p-galligan those are bag validation errors, which are expected and should be happening. You weren't seeing those before because the tests weren't calling that code...

**Review by p-galligan — APPROVED** [review]

**Review by HaSistrunk — APPROVED** [review]

File: aurora/bag_transfer/lib/transfer_routine.py [hunk 1]
```diff
@@ -103,37 +90,24 @@ def run_routine(self):
...
-                ["{} orgs were removed from routine".format(len(orgs_to_remove))]
-            )
+        self.active_organizations = [org for org in self.active_organizations if org not in orgs_to_remove]
 
     def build_contents_dictionary(self):
+        """Creates a dictionary of bags to process."""
         org_dir_contents = {}
         for org in self.active_organizations:
-            upload_dir = org.org_machine_upload_paths()[0]
+            upload_dir, _ = org.org_machine_upload_paths()
```
Comment (bonniegee) [inline aurora/bag_transfer/lib/transfer_routine.py:110]: Curious about why you changed the approach here--

**Review by bonniegee — APPROVED** [review]
Had a small question but look great!

File: aurora/bag_transfer/lib/transfer_routine.py (hunk 1 above)
Comment (helrond) [reply aurora/bag_transfer/lib/transfer_routine.py:110]: Eh, it's a preference thing but I find this a bit more readable than slicing.

**Pull request merged**